src/common/meta/src/wal/kafka/topic_manager.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use crate::error::Result;
+use crate::kv_backend::KvBackendRef;
+use crate::wal::kafka::topic::Topic;
+use crate::wal::kafka::topic_selector::{RoundRobinTopicSelector, TopicSelectorRef};
+use crate::wal::kafka::KafkaConfig;
+
+/// Manages topic initialization and selection.
+pub struct TopicManager {
+    topic_pool: Vec<Topic>,
+    topic_selector: TopicSelectorRef,
+    kv_backend: KvBackendRef,
+}
+
+impl TopicManager {
+    /// Creates a new topic manager.
+    pub fn new(config: &KafkaConfig, kv_backend: KvBackendRef) -> Self {
+        Self {
+            topic_pool: Vec::new(),
+            topic_selector: Arc::new(RoundRobinTopicSelector::new()),
```

Comment:
It seems like the `topic_selector` should be generated from the `config`

Reply:
Yes. The topic selector will be constructed according to the selector type specified in the wal config. But for now, the skeleton code does not implement such a logic.

Reply:
I am confused about this, why not implement it?